lib/dataModel.py

```diff
@@ -0,0 +1,198 @@
+        self.source = source
+        self.identifiers = []
+        self.instances = []
+        self.subjects = []
+        self.agents = []
+        self.links = []
+        self.measurements = []
+        self.license = None
+        self.language = None
+        self.title = None
+        self.subtitle = None
+        self.altTitle = None
+        self.rightsStatement = None
+        self.issued = None
+        self.published = None
+        self.medium = None
+        self.series = None
+        self.seriesPosition = None
+        self.primaryIdentifier = None
+
+    def addIdentifier(self, **identifierDict):
+        self.identifiers.append(Identifier.createFromDict(**identifierDict))
+
+    def addInstance(self, **instanceDict):
+        self.instances.append(InstanceRecord.createFromDict(**instanceDict))
+
+    def addSubject(self, **subjectDict):
+        self.subjects.append(Subject.createFromDict(**subjectDict))
+
+    def addAgent(self, **agentDict):
+        self.agents.append(Agent.createFromDict(**agentDict))
+
+    def addMeasurement(self, **measurementDict):
+        self.measurements.append(Measurement.createFromDict(**measurementDict))
+
+
+class InstanceRecord(DataObject):
+    def __init__(self, title=None, language=None):
+        super()
+        self.title = title
+        self.language = language
+        self.subtitle = None
+        self.altTitle = None
+        self.pubPlace = None
+        self.pubDate = None
+        self.edition = None
+        self.editionStatement = None
+        self.tableOfContents = None
+        self.copyrightDate = None
+        self.agents = None
+        self.identifiers = []
+        self.formats = []
+        self.measurements = []
+
+
+class Format(DataObject):
+    def __init__(self, contentType=None, link=None, modified=None):
+        super()
+        self.contentType = contentType
+        self.modified = modified
+        self.drm = None
+        self.measurements = []
+        self.rightsURI = None
+        self.link = None
+
+        if (isinstance(link, Link)):
+            self.link = link
+        else:
+            self.setLink(url=link)
+
+    def setLink(self, **linkFields):
+        newLink = Link.createFromDict(**linkFields)
+        self.link = newLink
+
+
+class Agent(DataObject):
+    def __init__(self, name=None, role=None, aliases=None, birth=None, death=None, link=None):
+        super()
+        self.name = name
+        self.sortName = None
+        self.lcnaf = None
+        self.viaf = None
+        self.biography = None
+        self.aliases = aliases
+        self.birthDate = birth
+        self.deathDate = death
+        self.link = link
+
+        if isinstance(role, (str, int)):
+            self.roles = [role]
+        else:
+            self.roles = role
+
+    # TODO This method is pretty ugly and there must be a better way to merge
+    # agent records. However, it does work
+    @staticmethod
+    def checkForMatches(newAgents, agents):
```

Comment:
If you're pulling agents from the `<author>` tags, my experience is that they always have LC and VIAF IDs. you can use those as a reliable way of determining whether two agents are the same.

If you're pulling agents from the "author" attribute of the `<work>` or `<edition>` tag, I recommend ignoring that data and waiting until you do the recursive lookup. In the recursive lookup, you'll have agent information in the more detailed `<author>` form.

Reply:
If you're trying to merge authors who VIAF thinks are different people... I just wouldn't do that automatically. Not without a lot of research. The VIAF dataset is the way it is for a reason, and even if a human being decides to override that, you should record all three pieces of information -- the two VIAF records and the human's opinion that they're the same person.

Reply:
The goal with this method was to merge authors from the Gutenberg (or other source) records, which don't have associated VIAF identifiers. However, I think from your other comments, attempting to do this at this stage is not a good idea and should be put off to a normalization step later on.

Definitely agree that merging VIAF records is always a bad idea, the goal here was to privilege data from OCLC over what we're getting from the source records.